desktop-app/app/actions/browser.js

```diff
@@ -698,6 +716,23 @@ export function onDeviceMutedChange(deviceId, isMuted) {
   };
 }
 
+export function onAllDevicesDesignModeChange() {
+  return (dispatch: Dispatch, getState: RootStateType) => {
+    const {
+      browser: {allDevicesInDesignMode},
+    } = getState();
+    const next = !allDevicesInDesignMode;
+    pubsub.publish(TOGGLE_DEVICE_DESIGN_MODE_STATE, [{designMode: next}]);
+    dispatch(toggleAllDevicesDesignMode(next));
+  };
+}
+
+export function onDeviceDesignModeChange(deviceId, designModeOn) {
```

Comment:
Since the functionality is a toggle of a given device's design mode, it should just take the `deviceId` as the parameter and use the state in the reducer to toggle the flag. This ensures that the consumer can only toggle as the functionality implies.

Reply:
that makes sense